CsvParser/csvParser.go

```diff
@@ -94,47 +98,151 @@ func CsvParser(filePath string, numIncidents ...int) []incident.DefaultIncident
+	gpStart, gpEnd := i.GetGoodPeriod()
+	bpStart, bpEnd := i.GetBadPeriod()
+	gMetric := i.GetGoodMetric()
+	bMetric := i.GetBadMetric()
+	severity := i.GetSeverity()
+	testsAffected := i.GetTestsAffected()
+	gpInfo := i.GetGoodPeriodInfo()
+	bpInfo := i.GetBadPeriodInfo()
+	iInfo := i.GetIncidentInfo()
+	locationString := i.GetLocation()
+	asn := i.GetASN()
+	inc := incident.IncidentData{gpStart, gpEnd, bpStart, bpEnd, gMetric, bMetric, asn, locationString, severity, testsAffected, gpInfo, bpInfo, iInfo}
+	return inc
 }
 
-func makeJsonObjFile(arr []incident.Incident) *os.File {
-	// numObjects determines how many incidents are stored in the json
-	const numObjects = 1
-	f, err := os.Create("incidents.json")
-	var objs [numObjects]incident.IncidentData
-	if err != nil {
-		log.Fatal(err)
-		return f
+// Takes in a location code and slices it into different location levels
+// Returns an array of strings, where each string represents a location
+func parseLocationCode(locationCode string) []string {
+	locationCodesArr := make([]string, 0)
+	upToStateLevelLocationLen := 6 // Examples: nausca, nausco
+
+	if (len(locationCode)) > upToStateLevelLocationLen {
+		// Increment by 2 because location code consists of two characters at each level
+		for i := 0; i < upToStateLevelLocationLen; i += 2 {
+			locationCodesArr = append(locationCodesArr, locationCode[i:i+2])
+		}
+
+		locationCodesArr = append(locationCodesArr, locationCode[upToStateLevelLocationLen:])
+
+		return locationCodesArr
 	}
 
-	for i := 0; i < numObjects; i++ {
-		gpStart, gpEnd := arr[i].GetGoodPeriod()
-		bpStart, bpEnd := arr[i].GetBadPeriod()
-		gMetric := arr[i].GetGoodMetric()
-		bMetric := arr[i].GetBadMetric()
-		severity := arr[i].GetSeverity()
-		testsAffected := arr[i].GetTestsAffected()
-		gpInfo := arr[i].GetGoodPeriodInfo()
-		bpInfo := arr[i].GetBadPeriodInfo()
-		iInfo := arr[i].GetIncidentInfo()
-		inc := incident.IncidentData{gpStart, gpEnd, bpStart, bpEnd, gMetric, bMetric, severity, testsAffected, gpInfo, bpInfo, iInfo}
-		objs[i] = inc
+	for i := 0; i < len(locationCode); i = i + 2 {
+		locationCodesArr = append(locationCodesArr, locationCode[i:i+2])
 	}
-	bytes, err := json.Marshal(objs)
-	n, err := f.Write(bytes)
 
-	if err != nil {
-		log.Fatal(n)
-		log.Fatal(err)
-		f.Close()
-		return f
+	return locationCodesArr
+}
+
+// Checks if a specific directory, which corresponds to a specific location, has been created
+// Returns a bool
+func dirExists(originPath string, dir string) bool {
+	actualPath := originPath + "/" + dir
+
+	if _, err := os.Stat(actualPath); !os.IsNotExist(err) {
+		return true
 	}
 
-	return f
+	return false
+}
+
+// Constructs incidents file hierarchy on basis of an incident location code
+// Dynamically construct a directory hierarchy and return the path where an incident ends on the disk 
+func dynamicallyMakeDir(originPath string, locationCode string) string {
```

Comment:
Can you add a comment at the top of this function that gives an example of how to call it?